plugins/gcloud/lib/samson_gcloud/tag_resolver.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+#
+# find the tag in the latest images, there is no search api for tags
+# https://cloud.google.com/sdk/gcloud/reference/container/images/list-tags
+module SamsonGcloud
+  class TagResolver
+    LIMIT = 1000 # number of images to check
+    TIMEOUT = 10 # s
+    DEFAULT_TAG = "latest"
+
+    class << self
+      def resolve_docker_image_tag(image)
+        return unless SamsonGcloud.gcr? image
+        return if image.match? Build::DIGEST_REGEX
+
+        tag = image.split(":", 2)[1] || DEFAULT_TAG
+
+        success, json = Samson::CommandExecutor.execute(
+          "gcloud", "container", "images", "list-tags", image, "--limit", LIMIT.to_s, "--format", "json",
```

Comment:
can't you use the container registry api directly rather than using something gcr specific?

Reply:
https://docs.docker.com/registry/spec/api/#listing-image-tags

Reply:
(that being said, using it against gcr requires docker auth credentials - not sure if you have them in samson or not already)

Reply:
it works, is easy to debug locally, reuses the gcloud executable we already use in different places, does not need new credentials ... so lots of upsides :)